### PR TITLE
Reading a recipe for the second time does not consume item

### PR DESCRIPTION
**Pull request by Veracity0**
1) In fact, if we didn't see you read it the first time, we can deduce you learned it.
2) Ash proxy record fields:
ash $item[warbear metalworking primer].skill -> Shrap
ash $item[Recipe of Before Yore: Pete's wily whey bar].recipe -> Pete's wiley whey bar
(Yes, I bug reported the disparity in names to KoL)
3) New String Modifier: Recipe: "CONCOCTION"
I only added them to Recipes of Before Yore, for now
4) When you find the Sot's Parcel, give a navigation link to go see him.
5) Also detect having learned recipes from 7 depleted grimicite items, 2 swagger shop items, and 2 pickpocketable items

### CODE REVIEW

**Review by midgleyc — DISMISSED** [review]
Is it worth adding modifiers for the other 11 [recipe items](https://kol.coldfront.net/thekolwiki/index.php/Category:Crafting_Plans) (minus the crystallized memory, which teaches multiple recipes)?

**Comment by Veracity0** [comment]
> Is it worth adding modifiers for the other 11 [recipe items](https://kol.coldfront.net/thekolwiki/index.php/Category:Crafting_Plans) (minus the crystallized memory, which teaches multiple recipes)?

I think it is. I'm not sure I want to add the detection for "you already know this recipe" unless it is the same. Heck, I could make it a negative test: unless we get "you learn a new recipe" assume you aleady learned it.

I'll take look.

**Comment by Veracity0** [comment]
I inverted the test - NOT seeing "You learn to craft a new item" - and added 11 more recipes.

**Review by midgleyc — APPROVED** [review]

File: src/net/sourceforge/kolmafia/request/UseItemRequest.java [hunk 1]
```diff
@@ -5877,6 +5877,42 @@ else if (responseText.contains("You can't figure out the angle")) {
...
+      case ItemPool.PLANS_FOR_GRIMACITE_GRAVY_BOAT:
+      case ItemPool.PLANS_FOR_GRIMACITE_WEIGHTLIFTING_BELT:
+      case ItemPool.PLANS_FOR_GRIMACITE_GRAPPLING_HOOK:
+      case ItemPool.PLANS_FOR_GRIMACITE_NINJA_MASK:
+      case ItemPool.PLANS_FOR_GRIMACITE_SHINGUARDS:
+      case ItemPool.PLANS_FOR_GRIMACITE_ASTROLABE:
+      case ItemPool.FETTUCINI_EPINES_INCONNU_RECIPE:
+      case ItemPool.SLAP_AND_SLAP_AGAIN_RECIPE:
+      case ItemPool.FUMBLE_FORMULA:
+      case ItemPool.MOTHERS_SECRET_RECIPE:
```
Comment (gausie) [inline src/net/sourceforge/kolmafia/request/UseItemRequest.java:5906]: @Veracity0 These changes are great but question: do we need to add to this switch now for every new recipe? Can we not derive this from parsing modifiers.txt?

**Pull request merged**